app/timetables/models.py

```diff
@@ -127,13 +133,15 @@ def calculate_cycle_day(self, date_datetime):
         days_interval = (date_datetime - self.ref_cycle_date).days
 
         if days_interval < 0:
-            cycle_day = None
-        else:
-            cycle_day = (((days_interval % self.cycle_length) + self.ref_cycle_day)
-                         % self.cycle_length)
+            raise ValidationError(
+                _('Supply a date later than %s' % (self.ref_cycle_date))
+            )
+
+        cycle_day = (((days_interval % self.cycle_length) + self.ref_cycle_day)
+                     % self.cycle_length)
 
```

Comment:
@andela-kerinoso In most parts of the code, we have been consistent with using the python's string.format method instead of the old `%` style. I think we should continue with the same pattern in this part of the code.